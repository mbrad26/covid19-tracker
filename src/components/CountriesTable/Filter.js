import React from 'react';

const Filter = ({
  query,
  reset,
  setQuery,
  globalData,
  setInactive,
  dataLoading,
  dataSuccess,
  zoomOnCountry,
  updateEndPoint,
  filteredCountries,
  historicalDataLoading,
  historicalCountryDataLoading,
  }) => {

  const handleChange = event => {
    setQuery(event.target.value);
  };

  console.log('FILTERED_COUNTRIES: ', filteredCountries);

  const handleSubmit = event => {
    event.preventDefault();
    if (filteredCountries.length === 1) {
      dataSuccess(filteredCountries[0]);
      zoomOnCountry(filteredCountries[0]);
      updateEndPoint(filteredCountries[0].country);
      historicalCountryDataLoading();
    } else {
      reset();
      setInactive();
      dataLoading();
      dataSuccess(globalData);
      historicalDataLoading();
    }
    setQuery('');
  };

  return (
    <div id='form'>
      <form data-testid='form' onSubmit={handleSubmit}>
        <input
          type='text'
          placeholder='Search ...'
          value={query}
          onChange={handleChange}
          autoFocus
        />
      </form>
    </div>
  )
}

export default Filter;
