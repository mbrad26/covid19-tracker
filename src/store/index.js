import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  applyMiddleware(saga, logger)
);

// console.log('STORE: ', store.getState());

saga.run(rootSaga);

export default store;
