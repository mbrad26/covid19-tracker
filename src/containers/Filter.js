import { connect } from 'react-redux';

import Filter from '../components/CountriesTable/Filter';
import { getData } from '../selectors/data';
import { doReset, doZoomOnCountry } from '../actions/mapboxData';
import { doDataLoading, doDataSuccess } from '../actions/globalData';
import {
  doUpdateEndPoint,
  doHistoricalDataLoading,
  doHistoricalCountryDataLoading,
} from '../actions/historicalData';

const mapStateToProps = ({ globalDataState }) => ({
  globalData: getData(globalDataState),
});

const mapDispatchToProps = dispatch => ({
  reset: () => dispatch(doReset()),
  dataLoading: () => dispatch(doDataLoading()),
  dataSuccess: data => dispatch(doDataSuccess(data)),
  zoomOnCountry: country => dispatch(doZoomOnCountry(country)),
  updateEndPoint: endPoint => dispatch(doUpdateEndPoint(endPoint)),
  historicalDataLoading: () => dispatch(doHistoricalDataLoading()),
  historicalCountryDataLoading: () => dispatch(doHistoricalCountryDataLoading()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Filter);
