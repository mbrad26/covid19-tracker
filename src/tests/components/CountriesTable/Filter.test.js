import React from 'react';
import { Provider } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';

import store from '../../../store';
import Filter from '../../../components/CountriesTable/Filter';

describe('Filter', () => {
  const filterProps = {
    query: '',
    reset: jest.fn(),
    setQuery: jest.fn(),
    globalData: {},
    setInactive: jest.fn(),
    dataLoading: jest.fn(),
    dataSuccess: jest.fn(),
    zoomOnCountry: jest.fn(),
    updateEndPoint: jest.fn(),
    filteredCountries: {},
    historicalDataLoading: jest.fn(),
    historicalCountryDataLoading: jest.fn(),
  };

  beforeEach(() => {
    render(
      <Provider store={store}>
        <Filter {...filterProps} />
      </Provider>
    );
  });

  it('renders snapshot', () => {
    const { container } = render (
      <Provider store={store}>
        <Filter {...filterProps} />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders all the props', () => {
    expect(screen.getByPlaceholderText('Search ...')).toBeInTheDocument();
  });

  it('on input change it calls handleChange', () => {
    fireEvent.change(screen.getByPlaceholderText('Search ...'), { target: { value: 'f' }});

    screen.debug();

    expect(filterProps.setQuery).toHaveBeenCalledWith('f');
  });
});
