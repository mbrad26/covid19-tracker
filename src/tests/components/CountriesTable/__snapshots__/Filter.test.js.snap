// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders snapshot 1`] = `
<div
  id="form"
>
  <form
    data-testid="form"
  >
    <input
      placeholder="Search ..."
      type="text"
      value=""
    />
  </form>
</div>
`;
